src/vec_buffers.h

```diff
@@ -255,27 +255,23 @@ Buffers<T>::Buffers(const Buffers<T>& vec1, const Buffers<T>& vec2)
 
     this->mem_alloc_case = BufMemAlloc::COMBINED;
 
-    this->mem = new std::vector<T*>();
-    this->mem->reserve(this->n);
-    this->mem->insert(
-        this->mem->end(), vec1.get_mem().begin(), vec1.get_mem().end());
-    this->mem->insert(
-        this->mem->end(), vec2.get_mem().begin(), vec2.get_mem().end());
+    mem.reserve(this->n);
+    mem.insert(mem.end(), vec1.get_mem().begin(), vec1.get_mem().end());
+    mem.insert(mem.end(), vec2.get_mem().begin(), vec2.get_mem().end());
 }
 
 template <typename T>
 Buffers<T>::~Buffers()
 {
-    if (this->mem_alloc_case != BufMemAlloc::NONE && this->mem != nullptr) {
+    if (this->mem_alloc_case != BufMemAlloc::NONE && mem.size() > 0) {
         if (this->mem_alloc_case == BufMemAlloc::FULL) {
             for (int i = 0; i < n; i++) {
-                aligned_deallocate<T>(this->mem->at(i));
+                aligned_deallocate<T>(mem[i]);
             }
         } else if (this->mem_alloc_case == BufMemAlloc::ZERO_EXTEND) {
             aligned_deallocate<T>(this->zeros);
         }
-        this->mem->shrink_to_fit();
-        delete this->mem;
+        mem.shrink_to_fit();
```

Comment:
Why calling `shrink_to_fit` here? (this may cause a reallocation and doesn't seems really useful here)